contrib/fedora/cjdns.nprocs.patch

```diff
@@ -0,0 +1,14 @@
+diff -up ./util/Security.c.nprocs ./util/Security.c
+--- ./util/Security.c.nprocs	2015-09-24 00:53:26.770308595 -0400
++++ ./util/Security.c	2015-09-24 00:53:46.003850640 -0400
+@@ -103,8 +103,8 @@ void Security_nofiles(struct Except* eh)
+ 
+ void Security_noforks(struct Except* eh)
+ {
+-    if (setrlimit(RLIMIT_NPROC, &(struct rlimit){ 0, 0 })) {
+-        Except_throw(eh, "Failed to set fork limit to 0 [%s]", strerror(errno));
++    if (setrlimit(RLIMIT_NPROC, &(struct rlimit){ 1, 1 })) {
++        Except_throw(eh, "Failed to set fork limit to 1 [%s]", strerror(errno));
```

Comment:
The function is already misnamed, as RLIMIT_NPROCS limits the number of *processes* running, not the number of forks.  Limiting the number of running processes to 0 happens to do what you want on recent linux kernels, as opposed to making setuid() fail (as it does on many kernels in production).  But this behaviour of NPROCS=0 is controversial, and you can't depend on it staying that way.  Setting NPROCS to 1 allows for the 1 running cjdns process, and lets setuid succeed on all linux versions - and prohibits forks as there is already one process.  

NOTE: there is also an interaction with using 'nobody' as the process owner, as having additional 'nobody' processes already running will again make setuid() fail.  That is another reason why you really need a unique owner for cjdroute (although it admittedly is difficult to accomplish portably).

Note that my first experience with cjdns was that it compiled up out of the box (v16), but failed to dropPermissions (still running as root) due to setting NPROCS to 0.